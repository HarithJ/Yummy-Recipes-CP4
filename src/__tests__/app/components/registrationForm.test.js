import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import RegistrationForm from '../../../components/auth/register/registrationForm.js'

describe("registration form component", () => {
  const registrationForm = shallow(<RegistrationForm />)

  it("renders properly", () => {
    expect(shallowToJson(registrationForm)).toMatchSnapshot();
  })

  it("has hide alert function", () => {
    expect(registrationForm.instance().hideAlert());
  })

  it("has a function to handle registration", () => {
    expect(registrationForm.instance().handleRegistration(global.event));
  })

  it("changes state", () => {
    registrationForm.setState({
      firstName: 'Harith',
      lastName: 'Bakhrani',
      userName: 'harithj',
      email: 'harithjaved@gmail.com',
      password: 'Password'
    });

    expect(registrationForm.find('#inputFirstName').props().value).toEqual('Harith');
    expect(registrationForm.find('#inputLastName').props().value).toEqual('Bakhrani');
    expect(registrationForm.find('#inputUsername').props().value).toEqual('harithj');
    expect(registrationForm.find('#inputEmail').props().value).toEqual('harithjaved@gmail.com');
    expect(registrationForm.find('#inputPassword').props().value).toEqual('Password');

  })
})
