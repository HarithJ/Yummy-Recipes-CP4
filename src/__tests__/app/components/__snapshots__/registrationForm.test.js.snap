// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration form component renders properly 1`] = `
<div
  className="container-fluid mTop"
>
  <h5
    className="text-primary"
  >
    Register Now!
  </h5>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group row"
    >
      <label
        className="col-sm-2 col-form-label"
        for="inputName"
      >
        Name*
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="inputFirstName"
          onChange={[Function]}
          placeholder="First name"
          type="text"
          value=""
        />
      </div>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="inputLastName"
          onChange={[Function]}
          placeholder="Last name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group row"
    >
      <label
        className="col-sm-2 col-form-label"
        for="userName"
      >
        User Name*
      </label>
      <div
        className="col"
      >
        <input
          className="form-control"
          id="inputUsername"
          onChange={[Function]}
          placeholder="User Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group row"
    >
      <label
        className="col-sm-2 col-form-label"
        for="inputEmail"
      >
        Email address*
      </label>
      <div
        className="col-sm-10"
      >
        <input
          aria-describedby="emailHelp"
          className="form-control"
          id="inputEmail"
          name="email"
          onChange={[Function]}
          placeholder="Enter email"
          type="email"
          value=""
        />
      </div>
      <small
        className="form-text text-muted col-sm-6"
        id="emailHelp"
      >
        We'll never share your email with anyone else.
      </small>
    </div>
    <div
      className="form-group row"
    >
      <label
        className="col-sm-2 col-form-label"
        for="inputPassword"
      >
        Password*
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="inputPassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="btn btn-primary"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
